app/services/inbound_webhooks/create_service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module InboundWebhooks
+  class CreateService < BaseService
+    def initialize(organization_id:, webhook_source:, payload:, event_type:, code: nil, signature: nil)
+      @organization_id = organization_id
+      @webhook_source = webhook_source
+      @code = code
+      @payload = payload
+      @signature = signature
+      @event_type = event_type
+
+      super
+    end
+
+    def call
+      inbound_webhook = InboundWebhook.create!(
+        organization_id:,
+        source: webhook_source,
+        code:,
+        payload:,
+        signature:,
+        event_type:
+      )
+
+      after_commit do
```

Comment:
It looks like we are not relying on transaction right? So I guess this `after_commit` block is not mandatory

Reply:
this ensures the job is not added to the queue and tried to be processed until the DB transaction is committed. This will avoid race condition between sidekiq and postgres where sidekiq picks the job but it is unable to find the resource in DB.

Reply:
I agree, but here `InboundWebhooks::CreateService` is called directly from the controller and no transaction is created in it so the job will always be called after the InboundWebhook creation transaction.
That's why I was asking. No a major issue anyway ;)